test/unit/frontend/verify-jira-middleware.test.ts

```diff
@@ -29,129 +47,235 @@ describe("#verifyJiraMiddleware", () => {
 			clientKey: "abc123",
 			enabled: true,
 			secrets: "def234",
-			sharedSecret: "ghi345",
+			sharedSecret: testSharedSecret,
 			subscriptions: jest.fn().mockResolvedValue([])
 		};
 	});
 
-	describe("GET request", () => {
-		const buildRequest = (jiraHost, secret = "secret"): any => {
-			const jwtValue = jwt.encode("test-jwt", secret);
-
-			return {
-				query: {
-					xdm_e: jiraHost,
-					jwt: jwtValue
-				},
-				session: {
-					jiraHost: subscription.jiraHost
-				},
-				addLogFields: jest.fn()
-			};
+	const buildRequest = (jiraHost, secret = "secret"): any => {
+		const jwtValue = encodeSymmetric({
+			qsh: "context-qsh",
+			iss: "jira"
+		}, secret);
+
+		return {
+			query: {
+				xdm_e: jiraHost,
+				jwt: jwtValue
+			},
+			session: {
+				jiraHost: subscription.jiraHost,
+				jwt: jwtValue
+			},
+			addLogFields: jest.fn(),
+			log: logger
 		};
+	};
 
-		it("should call next with a valid token and secret", async () => {
-			mocked(Installation.getForHost).mockResolvedValue(installation);
-			const req = buildRequest("test-host", "secret");
-			jwt.decode(req.query.jwt, "secret");
+	//Tests which should pass regardless if USE_JWT_SIGNED_INSTALL_CALLBACKS feature flag is on or off
+	const commonTests = () => {
```

Comment:
Not sure if this follows best javascript practices, but I extracted some tests in a function to repeat them with and without feature flag.

I'd like to know what is the best way to do it, but I'll leave it as is hence we'll remove it anyway after the feature flag cleanup.